lxd/instance/drivers/driver_qemu.go

```diff
@@ -3222,29 +3283,40 @@ func (d *qemu) addDriveConfig(bootIndexes map[string]int, driveConf deviceConfig
 
 		nodeName := fmt.Sprintf("%s%s", qemuBlockDevIDPrefix, escapedDeviceName)
 
-		permissions := unix.O_RDWR
+		if isRBDImage {
+			secretID := fmt.Sprintf("secret_%s_%s", blockDev["pool"], blockDev["user"])
```

Comment:
any max length of this key enforced by qemu btw? Should we also prefix with `pool` to avoid any conflicts? Also do we need to use `secret_` prefix as these are all secrets being added using `AddSecret` right?

Reply:
Does this work with hotplugging? What happens if the disk is removed whilst running, do we need to remove the secret, or if it changes if we re-add the disk does it update the secret?

Reply:
> any max length of this key enforced by qemu btw?

No idea. Need to check this.

> Should we also prefix with `pool` to avoid any conflicts? Also do we need to use `secret_` prefix as these are all secrets being added using `AddSecret` right?

We could just use `pool_` instead of `secret_`.

> Does this work with hotplugging?

Yes, it does.

> What happens if the disk is removed whilst running, do we need to remove the secret, or if it changes if we re-add the disk does it update the secret?

We don't actually remove the secret at all. That is because the secret is tied to a pool, not a volume/disk. However, we should update the secret if it changed.

Reply:
This hasn't been addressed yet either.